### ISSUE TITLE
Question: How to add virtual folders

### ISSUE THREAD

**Issue by wgebczyk** [indexzero/http-server#182]
When starting http-server, you define what is root folder.
How can I hack HS, and define additional virtual folders that are mapped to location outside of root tree?

**Comment by derhuerst** [comment]
AFAIK, `http-server` doesn't support this by itself, but you can use [symlinks](https://en.wikipedia.org/wiki/Symbolic_link#POSIX_and_Unix-like_operating_systems) for that.

**Comment by wgebczyk** [comment]
I don't want to use OS level hacks. I'm looking for place in HS code where URL part is mapped to physical location.

**Comment by derhuerst** [comment]
Why is that a hack? Why not use tried and mature techniques the operating system already provides?

**Comment by wgebczyk** [comment]
Because I don't want to tie working folder of repository with project. It might be possible to script link and unlink on start and stop of HS, but more simple approach would be to hack HS to allow this rather standard feature...

But I accept the fact on HS limitation.

**Comment by derhuerst** [comment]
Where is the difference between symlinking the folder and adding it as a virtual server?

**Comment by wgebczyk** [comment]
Not virtual server - just virtual folder. There are a few reasons:
* writing additional batch file to be executed before HS and after to cleanup
* maintaining 2 versions of that script for WIN and LINUX
* handle interrupted HS and cleanup and initialization I unclean env
* I don't want to have those folders to be visible as folders to not confuse Code Editors that there are sources duplicated
etc...

Having additional if or hashtable lookup in HS function that maps incoming URL to physical location seems to me much ore natural. Having located place where this mapping occurs and how to extend this functionality would be sufficient for me.

As I wrote. I accept the HS limitation and I'm using HS only in non WIN env. On WIN IIS Express gives me all what I want. In truth I've started with IIS EX and added option to test web site on non WIN platform recently.
